main.js

```diff
@@ -313,7 +323,27 @@ if (shouldQuit) {
 
 app.on('ready', () => {
   if ((process.env.NODE_ENV === 'production') || process.env.SLOBS_FORCE_AUTO_UPDATE) {
-    (new Updater(startApp)).run();
+    const updateInfo = {
+      baseUrl: 'https://s3-us-west-2.amazonaws.com/streamlabs-obs',
+      version: pjson.version,
+      exec: process.argv,
+      cwd: process.cwd(),
+      waitPid: [ process.pid ],
+      appDir: path.dirname(app.getPath('exe')),
+      tempDir: path.join(app.getPath('temp'), 'slobs-updater'),
+      cacheDir: app.getPath('userData'),
+      versionFileName: `${releaseChannel}.json`
+    };
+
+    console.log(updateInfo);
+    bootstrap(updateInfo).then((updating) => {
+      if (updating) {
+        console.log('Closing for update...');
+        app.exit();
+      } else {
+        startApp();
+      }
+    });
```

Comment:
I'm unsure if this does all that I intended but the exported function from the bootstrap module looks like this:
```
module.exports = async (info) => {
    return entry(info).catch((error) => {
        console.log(error.message);
        return Promise.resolve(false);
    });
}
```
The intention here is that if the entry promise is resolved, it's passed back to the application. Entry may also return false as a valid value as well! If an error or exception occurs in entry, it should be caught in the catch handler here and return false instead of passing that exception to the application.

Reply:
Yup that should do what you want.